adobe/react-spectrum
# Fix edge cases in DatePicker with min/max dates and eras

Fixes #3109.

This fixes a few edge cases:

* Minimum and maximum dates in a calendar system, e.g. we currently support up to year 9999 in the Gregorian calendar, and some calendar systems have minimum dates as well (e.g. we only support back to the Meiji era (~1868) in the Japanese calendar). Now we constrain the dates within the valid range of the calendar system.
* Dates in the BC era of the Gregorian calendar should display the era wherever they are shown.
* Noticed a weird bug in Firefox where sometimes clicking on a segment in DatePicker would not make it editable.

## 📝 Test Instructions:

1. In DatePicker, enter a date in December of the year 9999, and open Calendar. Verify that the next button is disabled. Verify that you cannot navigate to any future dates via keyboard, etc.
2. In the date picker, change the year to 1, and press the down arrow. Verify that you are now at 1 BC (the era should appear). 
3. Change the date to January of the year 9999 BC. Open the calendar, and verify that the previous button is disabled.
4. Change the date to September of 1868. Switch the calendar system to Japanese. Open the calendar, and verify that the previous button is disabled, and the month appears to start at day 8 (on a Tuesday). This is the first supported date.
5. With dates in BC, verify that all places the date is displayed or announced include the BC era.
